anta/tests/routing/bgp.py

```diff
@@ -1333,3 +1333,85 @@ def test(self) -> None:
             # Verify warning limit if provided. By default, EOS does not have a warning limit and `totalRoutesWarnLimit` is not present in the output.
             if warning_limit is not None and (actual_warning_limit := peer_data.get("totalRoutesWarnLimit", 0)) != warning_limit:
                 self.result.is_failure(f"{peer} - Maximum routes warning limit mismatch - Expected: {warning_limit}, Actual: {actual_warning_limit}")
+
+
+class VerifyBGPPeerPrefixes(AntaTest):
+    """Verifies the consistency of IPv4 prefixes in a BGP session.
+
+    This test performs the following checks for each specified address family:
+
+      1. Confirms that the specified VRF is configured.
+      2. Confirms all received prefixes were accepted.
+
+    Expected Results
+    ----------------
+    * Success: If the `PfxRcd` equals `PfxAcc`, indicating that all received prefixes were accepted.
+    * Failure: If any of the following occur:
+        - The specified VRF is not configured.
+        - The `PfxRcd` is not equal to `PfxAcc`, indicating that some prefix(s) were rejected/filtered out or peers are not configured.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.routing:
+      bgp:
+        - VerifyBGPPeerPrefixes:
+            address_families:
+              - afi: ipv4
+                safi: unicast
+                peers:
+                  - 10.100.0.8
+                  - 10.100.0.10
+              - afi: ipv4
+                safi: multicast
+              - afi: evpn
+              - afi: vpn-ipv4
+              - afi: ipv4
+                safi: labeled-unicast
+    ```
+    """
+
+    categories: ClassVar[list[str]] = ["bgp"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show bgp summary vrf all", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyBGPPeerPrefixes test."""
+
+        address_families: list[BgpAddressFamily]
+        """List of BGP address."""
```

Comment:
We should use `BgpPeer` model instead with the `capabilities` fields to know which AFI/SAFI we should check. You will have to update the field docstring and say it's also required in this test.
```suggestion
        bgp_peers: list[BgpPeer]
        """List of BGP IPv4 peers."""
```

Reply:
Also update the logic below. We should loop over each peer, check if it's present in its specified VRF, then you can loop over each capability and check if it's negotiated and `nlrisReceived` == `nlrisAccepted`.

Reply:
Updated. Thanks!!